pkg/kubevirt/validation/validation.go

```diff
@@ -81,9 +86,46 @@ func ValidateKubevirtProviderSpec(spec *api.KubeVirtProviderSpec) field.ErrorLis
 		}
 	}
 
+	if spec.Devices != nil {
+		disksPath := field.NewPath("devices").Child("disks")
+		disks := sets.NewString()
+
+		// +1 because of root-disk which is required and unique
+		volumesLen := len(spec.AdditionalVolumes) + 1
+
+		if disksLen := len(spec.Devices.Disks); disksLen > volumesLen {
+			errs = append(errs, field.Invalid(disksPath, disksLen, "the number of disks is larger than the number of volumes"))
+		}
+
+		for i, disk := range spec.Devices.Disks {
+			if disk.BootOrder != nil {
+				errs = append(errs, field.Forbidden(disksPath.Index(i).Child("bootOrder"), "cannot be set"))
+			}
+
+			if disk.Name == "" {
+				errs = append(errs, field.Required(disksPath.Index(i).Child("name"), "cannot be empty"))
+			} else if disks.Has(disk.Name) {
+				errs = append(errs, field.Invalid(disksPath.Index(i).Child("name"), disk.Name, "already exists"))
+				continue
+			} else if !hasDiskVolumeMatch(disk.Name, spec.AdditionalVolumes) && disk.Name != api.RootDiskName {
+				errs = append(errs, field.Invalid(disksPath.Index(i).Child("name"), disk.Name, "no matching volume"))
+			}
+			disks.Insert(disk.Name)
+		}
+	}
+
 	return errs
 }
 
+func hasDiskVolumeMatch(diskName string, volumes []api.AdditionalVolumeSpec) bool {
```

Comment:
This function checks whether the given slice has a volume with the given name.
```suggestion
func hasVolumeWithName(diskName string, volumes []api.AdditionalVolumeSpec) bool {
```